src/blockdata/witness.rs

```diff
@@ -282,18 +281,69 @@ impl serde::Serialize for Witness {
     where
         S: serde::Serializer,
     {
-        let vec: Vec<_> = self.to_vec();
-        serde::Serialize::serialize(&vec, serializer)
+        use hashes::hex::ToHex;
+        use serde::ser::SerializeSeq;
+
+        if serializer.is_human_readable() {
+            let mut seq = serializer.serialize_seq(Some(self.witness_elements))?;
+            for elem in self.iter() {
+                seq.serialize_element(&elem.to_hex())?;
+            }
+            seq.end()
+        } else {
+            let vec: Vec<_> = self.to_vec();
+            serde::Serialize::serialize(&vec, serializer)
+        }
```

Comment:
We should do the same `serialize_element` loop for non-human-readable to avoid `to_vec()`

```rust
let human_readable = serializer.is_human_readable();
let mut seq = serializer.serialize_seq(Some(self.witness_elements))?;

for elem in self.iter() {
    if human_readable {
        seq.serialize_element(&elem.to_hex())?;
    } else {
        seq.serialize_element(&elem)?;
    }
}
seq.end()
```

Reply:
Yes please, this solves https://github.com/rust-bitcoin/rust-bitcoin/issues/942

Reply:
Good idea, will do. Thanks